extensions/modules/file/src/main/java/org/exist/xquery/modules/file/Sync.java

```diff
@@ -55,183 +62,356 @@
+        }
+        return subCollections;
+    }
+
+    private void pruneCollectionEntries(
+            final Collection collection,
+            final String rootTargetAbsPath,
+            final Path targetDir,
+            final List<String> excludes,
+            final MemTreeBuilder output) {
+        try (final Stream<Path> fileStream = Files.walk(targetDir, 1)) {
+            fileStream.forEach(path -> {
+                try {
+                    // guard against deletion of output folder
+                    if (rootTargetAbsPath.startsWith(path.toString())) {
+                        return;
+                    }
+
+                    if (isExcludedPath(rootTargetAbsPath, path, excludes)) {
+                        return;
+                    }
+
+                    final String fileName = path.getFileName().toString();
+                    final XmldbURI dbname = XmldbURI.xmldbUriFor(fileName);
+                    final String currentCollection = collection.getURI().getCollectionPath();
+
+                    if (collection.hasDocument(context.getBroker(), dbname)
+                            || collection.hasChildCollection(context.getBroker(), dbname)
+                            || currentCollection.endsWith("/" + fileName)) {
+                        return;
+                    }
+
+                    // handle non-empty directories
+                    if (Files.isDirectory(path)) {
+                        deleteWithExcludes(rootTargetAbsPath, path, excludes, output);
+                    } else {
+                        Files.deleteIfExists(path);
+                        // reporting
+                        output.startElement(Sync.FileDeleteElement, null);
+                        output.addAttribute(Sync.FileAttribute, path.toAbsolutePath().toString());
+                        output.addAttribute(Sync.NameAttribute, fileName);
+                        output.endElement();
+                    }
+
+                } catch (IOException | URISyntaxException
+                        | PermissionDeniedException | LockException e) {
+                    reportError(output, e.getMessage());
+                }
+            });
+        } catch (IOException e) {
+            reportError(output, e.getMessage());
+        }
+    }
+
+    private void saveFile(final Path targetFile, final DocumentImpl doc, final Date startDate, final MemTreeBuilder output) throws LockException {
+        try (final ManagedLock<MultiLock[]> lock = context.getBroker().getBrokerPool().getLockManager().acquireDocumentReadLock(doc.getURI())) {
+            if (startDate == null || (
+                    doc.getLastModified() > startDate.getTime() && (
+                            !Files.exists(targetFile) ||
+                                    Files.getLastModifiedTime(targetFile).compareTo(FileTime.fromMillis(doc.getLastModified())) >= 0
+                    ))) {
+                output.startElement(Sync.FileUpdateElement, null);
```

Comment:
The static-final object members don't need to be be prefixed `Sync.` as they are already in scope.